wandb/sdk/launch/deploys/kubernetes/launch-agent.yaml

```diff
@@ -0,0 +1,52 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: launch-agent
+  namespace: wandb
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: launch-agent
+  template:
+    metadata:
+      labels:
+        app: launch-agent
+    spec:
+      serviceAccountName: wandb-launch-serviceaccount
+      containers:
+      - name: launch-agent
+        image: kaniko-build-test:v1
+        resources:
+          limits:
+            memory: "2Gi"
+            cpu: "1000m"
+        securityContext:
+          allowPrivilegeEscalation: false
+          runAsNonRoot: true
+          runAsUser: 1000
+          capabilities:
+            drop: ["ALL"]
+          seccompProfile:
+            type: RuntimeDefault
+        env:
+        - name: WANDB_API_KEY
+          valueFrom:
+            secretKeyRef:
+              name: wandb-api-key
+              key: password
+        volumeMounts:
+        - name: wandb-launch-config
+          mountPath: /home/launch_agent/.config/wandb
+          readOnly: true
+        - name: aws-secret # if you intend on using kaniko with ecr create a secret containing your aws credentials, with this name
+          mountPath: /home/launch_agent/.aws
+          readOnly: true
```

Comment:
I would imagine lot's of enterprise users will not like this.   Have we thought about using the instance metadata or whatever short lived credential logic that EKS prefers for getting cloud creds into containers?  Long lived secrets managed by the user are a big anti-pattern.  It's fine to support this, but we should think about the right way sooner than later.

Reply:
There is a separate `launch-agent.yaml` file that exists in a sibling directory that is for running the agent in an aws-eks instance. This just adds support for using kaniko with an aws ecr outside of eks.

Reply:
There is a separate launch-agent.yaml file that exists in a sibling directory that is for running the agent in an aws-eks instance. This just adds support for using kaniko with an aws ecr outside of eks.

Reply:
Kaniko will look in the environment for AWS credentials.  Hard coding an access key and secret key into the aws config file is an anti-pattern.  The prefered method is the instance metadata service, or short lived credentials provided by a k8s plugin.  This is the preferred way to get IAM credentials into k8s payloads for eks: https://docs.aws.amazon.com/eks/latest/userguide/iam-roles-for-service-accounts.html

Reply:
Maybe this is unclear, I'm having the user create their own `aws-secret` with the credentials outside of this file, we're just mounting it in. This is how the kaniko docs indicate it can be done here: https://github.com/GoogleContainerTools/kaniko

I'm happy to switch over to using an IAM role instead, but I'm tempted to get this solution into this release and work on that for the next release. I understand it's an anti pattern, but I'd really like to get this into next week's release

Reply:
I think in the aws-eks folder, this line is not needed. We should have it in the generic `kubernetes` config but here, we should recommend and rely on instance metadata for iam roles

Reply:
This does not exist in the aws-eks folder's launch-agent.yaml

Reply:
@KyleGoyette I'm fine with this for the initial release I'm just saying we need to be aware this is clunky and will likely get pushback so let's just track making the right way a first class citizen as well.  We'll always need this clunky way because minio for instance will likely need to use it.